cpp/src/binaryop/decimal_ops.cu

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/decimal/decimal_ops.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/transform.h>
+
+namespace cudf {
+namespace detail {
+
+template <typename DecimalType>
+struct divide_decimal_functor {
+  numeric::decimal_rounding_mode rounding_mode;
+
+  __device__ DecimalType operator()(DecimalType const& lhs, DecimalType const& rhs) const
+  {
+    return numeric::divide_decimal(lhs, rhs, rounding_mode);
+  }
+};
+
+std::unique_ptr<column> divide_decimal_impl(column_view const& lhs,
+                                            column_view const& rhs,
+                                            numeric::decimal_rounding_mode rounding_mode,
+                                            rmm::cuda_stream_view stream,
+                                            rmm::device_async_resource_ref mr)
+{
+  using namespace numeric;
```

Comment:
I'd like to avoid this `using namespace` if possible. It is easier to trace when namespaces are explicitly written.